frontend/src/features/userProfile/state/userProfileActions.ts

```diff
@@ -0,0 +1,78 @@
+import { ApiAction, ApiActionThunk, CALL_API } from "../../../store/api";
+
+import { NewSearchResult } from '../../../qri/search'
+import { UserProfileParams } from '../../../qri/userProfile'
+
+export const searchPageSizeDefault = 25
+
+export function loadUserProfile(username: string): ApiActionThunk {
+  return async (dispatch, getState) => {
+    return dispatch(fetchUserProfile(username))
+  }
+}
+
+export function loadUserProfileDatasets(username: string, userProfileParams: UserProfileParams): ApiActionThunk {
+  return async (dispatch, getState) => {
+    return dispatch(fetchUserProfileDatasets(username, userProfileParams))
+  }
+}
+
+export function loadUserProfileFollowing(username: string, userProfileParams: UserProfileParams): ApiActionThunk {
+  return async (dispatch, getState) => {
+    return dispatch(fetchUserProfileFollowing(username, userProfileParams))
+  }
+}
+
+function fetchUserProfile (username: string): ApiAction {
+  return {
+    type: 'userprofile',
+    [CALL_API]: {
+      endpoint: `identity/profile/${username}`,
+      method: 'GET',
+    }
+  }
+}
+
+const mapSearchResults = (results: any[]) => {
+  return results.map((d) => NewSearchResult(d))
+}
+
+const mapFrontendParams = (frontendParams: UserProfileParams) => {
+  // map frontend 'sort' param to backend 'orderBy'
+  return {
+    orderBy: frontendParams.sort === 'recentlyupdated' ? 'created,desc' : 'name,asc'
+  }
+}
+
+function fetchUserProfileDatasets (username: string, userProfileParams: UserProfileParams): ApiAction {
+
+  return {
+    type: 'userprofiledatasets',
+    [CALL_API]: {
+      endpoint: `dataset_summary/${username}`,
+      method: 'GET',
+      query: mapFrontendParams(userProfileParams),
+      pageInfo: {
+        page: userProfileParams.page,
+        pageSize: userProfileParams.pageSize
+      },
+      map: mapSearchResults
+    }
+  }
+}
+
+function fetchUserProfileFollowing (username: string, userProfileParams: UserProfileParams): ApiAction {
+  return {
+    type: 'userprofilefollowing',
+    [CALL_API]: {
+      endpoint: `follow/${username}`,
```

Comment:
probs more a note for backend & take it or leave it, but looking at this, I would expect an endpoint called `follow` to be how I follow a specific user, not showing me who that user follows.